dev/compile-map.py

```diff
@@ -7,49 +7,54 @@
+		'vbsp2': '${bin}/vbsp2 -game ${game} ${file}',
+	},
+	'final': {
+		'vvis': '${bin}/vvis -game ${game} ${bspfile}',
+		'vrad': '${bin}/vrad -game ${game} -noao -final -textureshadows -StaticPropLighting -threads $threads -hdr -StaticPropPolys ${bspfile}',
+		'vbsp': '${bin}/vbsp -game ${game} ${file}',
+		'vbsp2': '${bin}/vbsp2 -game ${game} ${file}',
+	}
+}
+
+profilers = {
+	'vtune': '$vtune -collect hotspots $exe',
+	'perf': 'perf record $exe'
+}
 
 argparser = argparse.ArgumentParser(description='Test harness for vrad, vvis, vbsp')
 argparser.add_argument('--profile-vrad', action='store_true')
 argparser.add_argument('--profile-vbsp', action='store_true')
 argparser.add_argument('--profile-vvis', action='store_true')
-argparser.add_argument('--profiler', type=str, choices=['vtune', 'perf'])
+argparser.add_argument('--profiler', type=str, choices=list(profilers.keys()))
 argparser.add_argument('--skip-vvis', action='store_true')
 argparser.add_argument('--vbsp2', action='store_true')
 argparser.add_argument('--skip-vbsp', action='store_true')
 argparser.add_argument('--skip-vrad', action='store_true')
 argparser.add_argument('--show-graph', action='store_true')
 argparser.add_argument('--threads', default=-1, type=int, help='Number of threads to use. Defaults to half of your systems core count')
-argparser.add_argument('--config', nargs='+', default=['fast'], help='Configs to use/compare')
+argparser.add_argument('--config', nargs='+', default=['fast'], choices=list(configs.keys()), help='Configs to use/compare')
 argparser.add_argument('--game', type=str, default='p2ce', choices=['p2ce', 'momentum'], help='Games to compile for')
 argparser.add_argument('--bench', action='store_true', help='Benchmark the compilers')
 argparser.add_argument('map', metavar='Map', type=str, nargs=1, help='Map to compile')
 args = argparser.parse_args()
 
-configs = {
-	'fast': {
-		'vvis': './bin/linux64/vvis -game $game -fast $bspfile',
-		'vrad': './bin/linux64/vrad -game $game -noao -StaticPropLighting -threads $threads -hdr -fast $bspfile',
-		'vbsp': './bin/linux64/vbsp -game $game $file',
-		'vbsp2': './bin/linux64/vbsp2 -game $game $file',
-	},
-	'normal': {
-		'vvis': './bin/linux64/vvis -game $game $bspfile',
-		'vrad': './bin/linux64/vrad -game $game -noao -textureshadows -StaticPropLighting -threads $threads -hdr $bspfile',
-		'vbsp': './bin/linux64/vbsp -game $game $file',
-		'vbsp2': './bin/linux64/vbsp2 -game $game $file',
-	},
-	'final': {
-		'vvis': './bin/linux64/vvis -game $game $bspfile',
-		'vrad': './bin/linux64/vrad -game $game -noao -final -textureshadows -StaticPropLighting -threads $threads -hdr -StaticPropPolys $bspfile',
-		'vbsp': './bin/linux64/vbsp -game $game $file',
-		'vbsp2': './bin/linux64/vbsp2 -game $game $file',
-	}
-}
-
-VTUNE_PATH="$vtune -collect hotspots $exe"
-PERF_PATH="perf record $exe"
 timers = []
-threads = multiprocessing.cpu_count() / 2
+threads = multiprocessing.cpu_count()
```

Comment:
isnt this usually - 1? im not sure if thats really necessary

Reply:
Eh, ninja defaults to `nproc`, I think -1 was just a me thing